### ISSUE TITLE
Redefining UI

### ISSUE THREAD

**Issue by Dzejzik** [cgeo/cgeo#7030]
Hey guys,
I've been using c:geo a few years ago a lot then I've stopped and today I've installed it again.
What a surprise for me was the same UI as before. In the evening I've got some contribution desire and I've sketch some ideas to improve UI of the app. The icons are Google material icons. I'm not a graphic designer but I work in automotive and my main focus is on wireframing and prototyping car UIs.

So if you like the concept feel free to contact me and I can make more screens or start work on html prototype that you can use as a guideline for the graphic designer and then for implementing UI into the app. 

![sum](https://user-images.githubusercontent.com/39681127/40591393-a2dd64c8-6210-11e8-919c-407de614ff95.png)

**Comment by Lineflyer** [comment]
Hi @Dzejzik and thanks for your proposal and example views.
We have quite a bunch of similar issues here, proposing a new UI or discussing the concept for such a changeover, e.g.: #5927 #4597 #6361 

For sure IMHO a design overhaul is required in c:geo, however we need someone able and willing to contribute to such activity. Up to now no one showed up, who would be able to start programming such an approach. Therefore continuing here will only make sense for now, if you could also help working on the implementation?!

**Comment by dcz-switcher** [comment]
Hello,

In my opinion, before dive into Android Studio (or Eclipse), we must adopt a user centric approach.

Asking for users what are they looking for in term of experience when playing geocaching, what are their objectives, pain points etc.
Then analysing the current c:geo app and competitors
and finally, define the key concepts of our design

For fun, I decided to do this and here is my current state

I pushed a survey in a French Facebook group to better understand how they play geocaching.
85 peoples answered

Then, I made an analysis of the current app and focus on this 3 flows
- onboarding
- find a cache on the live map
- log a visit

With the users's answers in mind, I started building a quick prototype in sketch app and export to MarvelApp

For my prototype I follow this design principles

- keep dark colors to distinguish c:geo from the others apps
- simple interface : direct access to main options without open a sub-menu
- main actions must be accessible with the thumb, so positioned on the bottom of the device

My contraints

- because it is an open source projet, it may be hard to imagine a complete redesign
- enhance users flow first, so keeping some current screens is not a problem
- I must be able to develop my own proposals

Before test it with real users, I need to meet at least 5 of them to make quality interviews

After that, I will made some tests and iterates to optimise my prototype

If you are curious about my first prototype which is not definitive because I did not made interviews for now, here it is

https://drive.google.com/file/d/1UChUP1a2V-YGppC1tmaDTRSSQKw8WHeH/view?usp=sharing

**Comment by jonas-koeritz** [comment]
I really like the user centric home screen of your concept @dcz-switcher!
I'm using the Live Map most of the time to explore my surroundings and would love to have the Cache pop-up replaced by some more modern idea like a bottom sheet. A good reference would be an app that every Android user should be familiar with: Google Maps.
A bottom sheet with several information items and actions, in a c:geo color theme would do great here.

**Comment by rsudev** [comment]
@jonas-koeritz Thanks for the comment that led me to take a look again here.
@dcz-switcher Redesigning the flow to better support the main use-cases is an approach I appreciate.
Could you try to extend the idea to cover the case, that c:geo is not or not only used with gc.com? There are regions where (AFAIK) a regional service is much stronger than gc.com (oc.pl, gc.su) or additional services are used in addition (a number of people here use oc.de in addition or as the only option)

**Comment by jonas-koeritz** [comment]
There are many things to consider when implmenting that design.

- If there are multiple avatar images available, which one should c:geo choose?
- How should the various found caches counters be displayed?

I'm rather new here but I think there should be a UI-Overhaul project where all those thoughts and Ideas are collected.

**Comment by Lineflyer** [comment]
> I'm rather new here but I think there should be a UI-Overhaul project where all those thoughts and Ideas are collected.

Any proposal for a method or tool to do this?
In the past it was always "a problem" that we have several issues with bunch of ideas behind, but no good way to bring it into a final concept. You also have to consider that the team is not stable, but consists of more or less volunteer developers over time each having expertise in some areas but probably not enough knowledge and/or time to just do the whole thing alone at once.

So either we divide it into reasonable evolution steps or have to find some other way to track it.
Recently we also tried to get a Android design expert from some specialized platforms, who could take the lead of this (for free). Until now, not a big response to that.

On the other hand I always get the feedback, that most users like the current design despite the fact, that its not modern at all. Maybe we have to accept, that we cannot just redesign everything from scratch with our manpower.

**Comment by jonas-koeritz** [comment]
Maybe instead of a UI redefinition, as many users like it the way it is, there should be subtile changes, like:

- animation/transitions
- updated icons
- some redesigned/restructured dialogs
- Implementing some parts of the app as Fragments instead of seperate Activities
- ...

**Comment by moving-bits** [comment]
IMHO "redefining UI" has two sides:
- improving the user experience (as mentioned above, with many possible aspects and solutions) - should follow some (generally accepted) user style guide
- clearing up the internal (technical) workings (refactoring, upgrading to recent technical standards and procedures for Android programming)

**Comment by dcz-switcher** [comment]
The improvement of the experience depends on several elements, but user journeys play a major role.
When the user starts the application for the first time, he must be guided to perform the first necessary action: authentication on one or more geochaching platforms.
This done, it must be redirected immediately to the home screen to start playing, because this is the main goal.
Here is a proposal for the onboarding phase.
I have tried to keep the current screens as much as possible to minimize the development load required.
The first image is an analysis of the existing one and the second a proposal of a user journey

In addition to the user paths, the interface elements must be coherent and respect ergonomic principles. I propose some tracks like to emphasize the buttons of action with a background of color.
This takes us to the need to define a set of graphical and ergonomic rules to deliver the best experience.
In summary and in order, I would put: the courses, the ergonomics of the interfaces and a system design

![Onboarding Actual](https://user-images.githubusercontent.com/299807/58339683-65ac9680-7e4a-11e9-869e-adbd9f358a9c.jpg)

![Onboarding Proposal](https://user-images.githubusercontent.com/299807/58339557-27af7280-7e4a-11e9-8a59-078ad68b8db2.jpg)

**Comment by dcz-switcher** [comment]
For the tool, one of the best is probably [Sketch](https://www.sketch.com), but it is not free.
A great alternative is [Figma](https://www.figma.com) a web competitor of Sketch and it is may be more adapted to a collaborative project like c:geo. The starting plan is free for 3 team members.

**Comment by Tobii42** [comment]
> ![Onboarding Proposal](https://user-images.githubusercontent.com/299807/58339557-27af7280-7e4a-11e9-8a59-078ad68b8db2.jpg)

Hey there,

I hope it's okay to comment your (great!) work from a mostly user perspective with some development background. I like the simplification of the login process you propose. When thinking through the steps I found it a little unintuitive to press the back button after a successful login to start playing (or adding another account). I thought about how to make this a little easier and think that an even more shrunk-down login process (or onboarding process) would be better. My suggestion would be to go straight to the "start-playing" screen after a successful login and avoid showing the profile or the list of other services to login to.

Why would I do this so? I assume that the login to multiple accounts is more a pro-user feature. New players who want to play as quick as possible don't want or need multiple accounts and may even be confused by the possibility to login to more than one account. Pro-users on the other hand who want to play with multiple accounts already know what they are doing and will search for a multi-account login in the settings.

Additionally, viewing the account details right after login is not that important, I think. Do you do this in other apps? I cannot remember an app where that is the case. Normally you go to the functionality directly after logging in. This would avoid the need to press the "back" button.

**Comment by dcz-switcher** [comment]
Thanks @Tobii42 for your - greats - comments. You are right, it is not logic to use "back button" to continue.
Instead removing the possibility to connect multiple accounts (an expert's feature), we may keep it but remove the account's detail display after successful login.
What do you think of this updated version ?

Note : I added a screen before the permissions process to explain why c:geo need these.


![Onboarding Proposal _without back after login](https://user-images.githubusercontent.com/299807/62073296-ba043600-b240-11e9-8638-9fa845d20c84.jpg)

The Figma file was updated too
https://www.figma.com/file/UxSM7MklYqclsyduCTV6gTeA/cgeo-onboarding?node-id=0%3A1

**Comment by rsudev** [comment]
I like your last proposal, especially that it still allows to select
- more than one platform
- start without an account (pure offline usage is a valid use-case for c:geo for some)
while making the onboarding a lot easier.
Also a clean (visual) integration of the permission requests into this flow makes the technical implementation a lot easier.

Nevertheless is the onboarding IMO not the most crucial thing to look at (even though it is the easiest one), as you are a new user only a very small percentage of the time you interact with the app - and we don't need to have primarily an easy onboarding to 'sell' it.

But it can definitely be the starting point...
